circleguard/widgets.py

```diff
@@ -765,3 +771,24 @@ def __init__(self, text):
         layout.addWidget(QLabel(text), 0, 1, 1, 1)
 
         self.setLayout(layout)
+
+
+class EntryWidget(QFrame):
+    button_pressed = pyqtSignal(object)
+    """
+    Represents a single entry, consisting of a title, an action name and data which is stored for ease of use.
+    """
+    def __init__(self, title, action_name, data=None):
+        super().__init__()
+        self.data = data
+        self.button = QPushButton(action_name)
+        self.button.setFixedWidth(100)
+        self.button.clicked.connect(self._pressed_action)
+        layout = QGridLayout()
+        layout.addWidget(QLabel(title), 0, 0, 1, 1)
+        layout.addWidget(self.button, 0, 1, 1, 1)
+        layout.setContentsMargins(0, 0, 0, 0)
+        self.setLayout(layout)
+
+    def _pressed_action(self, _):
```

Comment:
button_pressed is better

Reply:
function `button_pressed` to emit a signal called `button_pressed`? I guess changing the signal name would work but I don't have a better name for it...

Reply:
signals should end in _signal